apis/python/src/tiledbsoma/soma_dataframe.cc

```diff
@@ -155,6 +155,17 @@ void load_soma_dataframe(py::module& m) {
             "maybe_soma_joinid_maxshape",
             &SOMADataFrame::maybe_soma_joinid_maxshape)
         .def_property_readonly(
-            "tiledbsoma_has_upgraded_domain", &SOMAArray::has_current_domain);
+            "tiledbsoma_has_upgraded_domain", &SOMAArray::has_current_domain)
+
+        .def(
+            "resize_soma_joinid",
+            [](SOMADataFrame& sdf, int64_t newshape) {
+                try {
+                    sdf.resize_soma_joinid(newshape);
+                } catch (const std::exception& e) {
+                    throw TileDBSOMAError(e.what());
+                }
+            },
+            "newshape"_a);
```

Comment:
I think it would be best to keep these `TileDBSOMAError` rethrows in the C++ code so that it's consistent for both R and Python.

Reply:
We found on #2963 and #2980 that this is absolutely (and unfortunately, but absolutely) necessary